src/common/datasource/src/lister.rs

```diff
@@ -76,7 +76,16 @@ impl Lister {
                         path: &file_full_path,
                     },
                 )?;
-                Ok(vec![Entry::new(&file_full_path)])
+                // Safety: file must exists
```

Comment:
I don't think the safety can be upheld. The file can be removed just in between with the two object store operations (`stat` and `list_with`).